facia-tool/public/js/widgets/columns/fronts-search.js

```diff
@@ -0,0 +1,72 @@
+import ko from 'knockout';
+import _ from 'underscore';
+import ColumnWidget from 'widgets/column-widget';
+import CONST from 'constants/defaults';
+import generateCollections from 'utils/generate-collections';
+
+export default class SearchConfig extends ColumnWidget {
+    constructor(params, element) {
+        super(params, element);
+        this.searchTerm = ko.observable('');
+        this.searchedFronts = ko.observableArray([]);
+        var originalFronts = _.reduce(JSON.parse(JSON.stringify(this.baseModel.frontsList())), function (frontList, front) {
```

Comment:
I don't like `JSON.parse(JSON.stringify` here. I think it's more readable if you reduce `this.baseModel.frontsList()` and then clone `front` inside the callback.
If you want to clone a nested object you can use `utils/clean-clone` otherwise a simple `_.extend` works fine.

You don't necessarily have to keep the same `front` structure, you can create a new object that better suits your needs, for instance you can do `front.id.toLowerCase()` here instead of having to do it on every search.

Keep in mind that originalFronts changes with time and can be updated while the app is running. For instance if you create a new front or someone else does it, you should still be able to find those through a search.
You can subscribe to `vars.model.state()` or maybe there's already a notification mechanism, I can't remember.